### PR TITLE
feat: add buttons states

### LINKED ISSUE #76: Transfer Action Button Component
## Implementation details

We need button that has multiple states, including loading and disabled state.

States:
image
image
image
image
image

### PR DESCRIPTION
**Pull request by Lykhoyda**
Closes #76 
- Add button component
- moved components to the common folder
- update ESlint

### CODE REVIEW

File: packages/widget/src/components/common/buttons/button.styles.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    &:active,
+    &:hover {
+      filter: brightness(140%);
+    }
+
+    &.disabled {
+      color: var(--zinc-400, #a1a1aa);
+      background: var(--zinc-200, #e4e4e7);
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.styles.ts:42]: do we need the fallback color?

File: packages/widget/src/components/common/buttons/button.styles.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+export const buttonStyles = css`
+  .button {
+    width: 100%;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0.75rem 1.25rem;
+    border-radius: 1rem;
+    background: var(--zinc-900, #18181b);
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.styles.ts:11]: do we need the fallback colors?

File: packages/widget/src/components/common/buttons/button.styles.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      color: var(--zinc-400, #a1a1aa);
+      background: var(--zinc-200, #e4e4e7);
+
+      &:hover {
+        filter: brightness(80%);
+      }
+    }
+
+    &.loading {
+      color: var(--zinc-200, #e4e4e7);
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.styles.ts:50]: same question as above

File: packages/widget/src/components/common/buttons/button.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+export class Button extends BaseComponent {
+  static styles = buttonStyles;
+
+  @property({})
+  text: string = '';
+
+  @property({ type: Object })
+  onClick: () => void = () => {};
+
+  @property({ type: Boolean, reflect: true })
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.ts:21]: I wonder why the reflect true? 🤔

**Review by wainola — CHANGES_REQUESTED** [review]

File: packages/widget/src/components/address-input/address-input.ts [hunk 5]
```diff
@@ -5,7 +5,8 @@ import { ifDefined } from 'lit/directives/if-defined.js';
 import { Network } from '@buildwithsygma/sygma-sdk-core';
 import { when } from 'lit/directives/when.js';
 import { validateAddress } from '../../utils';
-import { BaseComponent } from '../base-component/base-component';
+
+import { BaseComponent } from '../common/base-component/base-component';
```
Comment (wainola) [inline packages/widget/src/components/address-input/address-input.ts:9]: can we use named exports from some `index.ts` file? In doing so we avoid having `../common/base-component/base-component` and hence we would have `../common/base-component` as the url for the import? Wdty?

File: packages/widget/src/components/common/buttons/button.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    const buttonClasses = classMap({
+      button: true,
+      disabled: this.disabled,
+      loading: this.isLoading
+    });
+
+    return html`<button
+      type="button"
+      ?disabled="${this.disabled}"
+      @click=${ifDefined(
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.ts:41]: did the linter ask you to add the `ifDefined` directive? 🤔

File: packages/widget/src/components/common/buttons/button.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import { customElement, property } from 'lit/decorators.js';
+import { html } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { when } from 'lit/directives/when.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { BaseComponent } from '../base-component/base-component';
+import { loaderIcon } from '../../../assets';
+import { buttonStyles } from './button.styles';
+
+@customElement('sygma-button')
```
Comment (wainola) [inline packages/widget/src/components/common/buttons/button.ts:11]: perhaps we should use a more semantic name, since this is not a plain button, but rather one that reflect to loading state

File: packages/widget/tests/unit/components/common/buttons/button.test.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    const el = document.createElement('sygma-button');
+    assert.instanceOf(el, Button);
+  });
+
+  it('displays the button text', async () => {
+    const el = await fixture<Button>(
+      html`<sygma-button text="Click me"></sygma-button>`
+    );
+    assert.equal(el.text, 'Click me');
+    assert.include(el.shadowRoot?.textContent || '', 'Click me');
```
Comment (wainola) [inline packages/widget/tests/unit/components/common/buttons/button.test.ts:21]: why the or clause on the assertion?

**Review by wainola — CHANGES_REQUESTED** [review]

File: packages/widget/src/components/common/buttons/button.ts (hunk 7 above)
Comment (mpetrunic) [reply packages/widget/src/components/common/buttons/button.ts:11]: sygma-action-button?

File: packages/widget/src/components/common/buttons/button.styles.ts (hunk 1 above)
Comment (Lykhoyda) [reply packages/widget/src/components/common/buttons/button.styles.ts:42]: nope 😅

File: packages/widget/src/components/common/buttons/button.ts (hunk 4 above)
Comment (Lykhoyda) [reply packages/widget/src/components/common/buttons/button.ts:21]: I thought about styling the attribute similar to https://lit.dev/docs/components/properties/#:~:text=Permalink%20to%20%E2%80%9CEnabling,%23

But used the class in the end. So we can remove it to not overcomplicate the code

File: packages/widget/src/components/address-input/address-input.ts (hunk 5 above)
Comment (Lykhoyda) [reply packages/widget/src/components/address-input/address-input.ts:9]: It's a good idea and I tried to do so. But somehow re-export for abstract classes doesn't work. Looks like this issue has existed for quite a long time

File: packages/widget/src/components/common/buttons/button.ts (hunk 6 above)
Comment (Lykhoyda) [reply packages/widget/src/components/common/buttons/button.ts]: removed, actually lint ask to remove it

File: packages/widget/src/components/amount-selector/amount-selector.ts [hunk 9]
```diff
@@ -132,20 +131,20 @@ export class AmountSelector extends BaseComponent {
               class="amountSelectorInput"
               placeholder="0.000"
               @change=${this._onInputAmountChangeHandler}
-              value=${this.amount || ifDefined(this.preselectedAmount)}
+              value=${this.amount ||
+              ifDefined(this.preselectedAmount).toString()}
```
Comment (mpetrunic) [inline packages/widget/src/components/amount-selector/amount-selector.ts:135]: Can you rollback lint changes? It will cause a massive amount of conflicts, it should be in separate PR.

**Review by mpetrunic — CHANGES_REQUESTED** [review]
